File: TreeViewAdvDemo/C#/Form1.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Windows.Forms;
using Syncfusion.Windows.Forms.Tools;

namespace TreeViewCustomControl
{
    public partial class Form1 : Form
    {
        #region Constructor

        public Form1()
		{
			InitializeComponent();
			this.treeViewAdv1.Nodes[0].CustomControl = this.comboBoxAdv1;
            this.treeViewAdv1.Nodes[2].Nodes[0].CustomControl = this.comboBoxAdv3;

            this.treeViewAdv1.Nodes[0].Tag = "Search node";
            this.treeViewAdv1.Nodes[1].Tag = "Search node";
            this.treeViewAdv1.Nodes[2].Tag = "Search node";
            this.treeViewAdv1.Nodes[2].Nodes[0].Tag = "Search node";

            //To search all nodes
            nodes = treeViewAdv1.SearchTree().Where(n => n.Tag == "Search node").ToList();
		}

        #endregion

        List<TreeNodeAdv> nodes;

        private void textBox1_TextChanged(object sender, EventArgs e)
        {
            this.listBox1.Items.Clear();
            if (this.textBox1.Text == "Search node")
            {
                foreach (TreeNodeAdv item in nodes)
                {
                   this.listBox1.Items.Add(item.Text + " :" + "Tag : " + item.Tag);
                }
            }
        }
	}

    public static class SOExtension
    {
        public static IEnumerable<TreeNodeAdv> SearchTree(this TreeViewAdv treeView)
        {
            return SearchTree(treeView.Nodes);
        }

        public static IEnumerable<TreeNodeAdv> SearchTree(this TreeNodeAdvCollection coll)
        {
            return coll.Cast<TreeNodeAdv>().Concat(coll.Cast<TreeNodeAdv>().SelectMany(x => SearchTree(x.Nodes)));
        }
    }
}

File: TreeViewAdvDemo/VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Windows.Forms
Imports Syncfusion.Windows.Forms.Tools

Namespace TreeViewCustomControl
    Partial Public Class Form1
        Inherits Form

#Region "Constructor"

        Public Sub New()
            InitializeComponent()
            Me.treeViewAdv1.Nodes(0).CustomControl = Me.comboBoxAdv1
            Me.treeViewAdv1.Nodes(2).Nodes(0).CustomControl = Me.comboBoxAdv3

            Me.treeViewAdv1.Nodes(0).Tag = "Search node"
            Me.treeViewAdv1.Nodes(1).Tag = "Search node"
            Me.treeViewAdv1.Nodes(2).Tag = "Search node"
            Me.treeViewAdv1.Nodes(2).Nodes(0).Tag = "Search node"

            'To search all nodes
            nodes = treeViewAdv1.SearchTree().Where(Function(n) n.Tag = "Search node").ToList()
        End Sub

#End Region

        Private nodes As List(Of TreeNodeAdv)

        Private Sub textBox1_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles textBox1.TextChanged
            Me.listBox1.Items.Clear()
            If Me.textBox1.Text = "Search node" Then
                For Each item As TreeNodeAdv In nodes
                    Me.listBox1.Items.Add(item.Text & " :" & "Tag : " & item.Tag)
                Next item
            End If
        End Sub
    End Class

    Public Module SOExtension
        <System.Runtime.CompilerServices.Extension()>
        Public Function SearchTree(ByVal treeView As TreeViewAdv) As IEnumerable(Of TreeNodeAdv)
            Return SearchTree(treeView.Nodes)
        End Function

        <System.Runtime.CompilerServices.Extension()>
        Public Function SearchTree(ByVal coll As TreeNodeAdvCollection) As IEnumerable(Of TreeNodeAdv)
            Return coll.Cast(Of TreeNodeAdv)().Concat(coll.Cast(Of TreeNodeAdv)().SelectMany(Function(x) SearchTree(x.Nodes)))
        End Function
    End Module
End Namespace